### ISSUE TITLE
Allow coordination controllers to modify the scheduling state of resources.

### ISSUE THREAD

**Issue by jmprusi** [kcp-dev/kcp#1506]
Some components require the ability to modify the scheduling of a resource, which requires access to the "state.workload.kcp.dev/<SyncTarget>" label. 

- The sync label should be protected to avoid users that are not part of a specific group (ex: `system:kcp:workload:coordination`) to modify its value. 

- This should allow the members of the group to modify the label "state.workload.kcp.dev/<SyncTarget>" value, not adding/removing new labels targeting different SyncTargets.

**Comment by jmprusi** [comment]
Some extra context: https://kubernetes.slack.com/archives/C021U8WSAFK/p1657635523445049

**Comment by sttts** [comment]
That being a group is not set in stone. Maybe we will have a `ThingCoordination`resource for coordinated to register for certain resources (plus a service account where we do a SAR against ThingCoordination for), or we have a `ThingWorkloadResource` where we do a `verb=coordinate` SAR against.

**Comment by sttts** [comment]
This currently falls under `reservedmetadata` admission, i.e. only `system:master` users/controllers can change the sync labels. Hence, we are currently on the safe side and this issue is about opening it up.

**Comment by jmprusi** [comment]
@sttts yes! after the `reservedmetadata` changes we are safe, as you say, now we should add a mechanism to for allow external actors to modify this label.
